packages/flutter/test/material/snack_bar_test.dart

```diff
@@ -685,6 +685,162 @@ void main() {
     expect(textTopRight.dy - snackBarTopRight.dy, padding);
   });
 
+  group('Snackbar constraints can be customized', () {
+    testWidgets('minHeight not ignore', (WidgetTester tester) async {
+      const double minHeight = 100;
+      await tester.pumpWidget(
+        MaterialApp(
+          home: Scaffold(
+            body: Builder(
+              builder: (BuildContext context) {
+                return GestureDetector(
+                  onTap: () {
+                    ScaffoldMessenger.of(context).showSnackBar(
+                      const SnackBar(
+                        content: Text('I am a snack bar.'),
+                        constraints: BoxConstraints(minHeight: minHeight),
+                      ),
+                    );
+                  },
+                  child: const Text('X'),
+                );
+              },
+            ),
+          ),
+        ),
+      );
+
+      await tester.tap(find.text('X'));
+      await tester.pump(); // start animation
+      await tester.pump(const Duration(milliseconds: 750));
+
+      final Finder materialFinder = find.descendant(
+        of: find.byType(SnackBar),
+        matching: find.byType(Material),
+      );
+      final Size size = tester.getSize(materialFinder);
+      expect(size.height, minHeight);
+    });
+
+    testWidgets('maxHeight not ignore', (WidgetTester tester) async {
+      const double maxHeight = 100;
+      await tester.pumpWidget(
+        MaterialApp(
+          home: Scaffold(
+            body: Builder(
+              builder: (BuildContext context) {
+                return GestureDetector(
+                  onTap: () {
+                    ScaffoldMessenger.of(context).showSnackBar(
+                      const SnackBar(
+                        content: SizedBox(
+                          height: maxHeight + 1,
+                          child: Text('I am a snack bar.'),
+                        ),
+                        constraints: BoxConstraints(maxHeight: maxHeight),
+                      ),
+                    );
+                  },
+                  child: const Text('X'),
+                );
+              },
+            ),
+          ),
+        ),
+      );
+
+      await tester.tap(find.text('X'));
+      await tester.pump(); // start animation
+      await tester.pump(const Duration(milliseconds: 750));
+
+      final Finder materialFinder = find.descendant(
+        of: find.byType(SnackBar),
+        matching: find.byType(Material),
+      );
+      final Size size = tester.getSize(materialFinder);
+      expect(size.height, maxHeight);
+    });
+
+    testWidgets('minWidth ignore', (WidgetTester tester) async {
```

Comment:
```suggestion
    testWidgets('constraints.minWidth is ignored if width is not null and less than constrains.minWidth', (WidgetTester tester) async {
```

Reply:
Could you also add two more unit tests to show that `constraints.minWidth` and `constraints.maxWidth` are respected when `width` is null?